Connection cleanup can trigger exception

**Describe the bug**
Debitor on stream close not being released and removing streams not related to the connection.

```
2023-09-24 10:07:07 started
2023-09-24 10:08:00 stopped
2023-09-24 10:08:00 org.agrona.concurrent.AgentTerminationException: java.lang.NullPointerException: Cannot invoke "java.util.function.LongConsumer.accept(long)" because "flusher" is null
2023-09-24 10:08:00     at io.aklivity.zilla.runtime.engine/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.doWork(DispatchAgent.java:681)
2023-09-24 10:08:00     at org.agrona.core/org.agrona.concurrent.AgentRunner.doDutyCycle(AgentRunner.java:291)
2023-09-24 10:08:00     at org.agrona.core/org.agrona.concurrent.AgentRunner.run(AgentRunner.java:164)
2023-09-24 10:08:00     at java.base/java.lang.Thread.run(Thread.java:1623)
2023-09-24 10:08:00 Caused by: java.lang.NullPointerException: Cannot invoke "java.util.function.LongConsumer.accept(long)" because "flusher" is null
2023-09-24 10:08:00     at io.aklivity.zilla.runtime.binding.kafka/io.aklivity.zilla.runtime.binding.kafka.internal.budget.KafkaMergedBudget.flush(KafkaMergedBudget.java:182)
2023-09-24 10:08:00     at io.aklivity.zilla.runtime.binding.kafka/io.aklivity.zilla.runtime.binding.kafka.internal.budget.KafkaMergedBudget.credit(KafkaMergedBudget.java:66)
2023-09-24 10:08:00     at io.aklivity.zilla.runtime.binding.kafka/io.aklivity.zilla.runtime.binding.kafka.internal.budget.KafkaMergedBudgetCreditor.credit(KafkaMergedBudgetCreditor.java:67)
2023-09-24 10:08:00     at io.aklivity.zilla.runtime.binding.kafka/io.aklivity.zilla.runtime.binding.kafka.internal.stream.KafkaClientConnectionPool$KafkaClientConnection.onConnectionWindow(KafkaClientConnectionPool.java:1280)
2023-09-24 10:08:00     at io.aklivity.zilla.runtime.binding.kafka/io.aklivity.zilla.runtime.binding.kafka.internal.stream.KafkaClientConnectionPool$KafkaClientConnection.onConnectionMessage(KafkaClientConnectionPool.java:1110)
2023-09-24 10:08:00     at io.aklivity.zilla.runtime.engine/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.handleReadInitial(DispatchAgent.java:1070)
2023-09-24 10:08:00     at io.aklivity.zilla.runtime.engine/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.handleRead(DispatchAgent.java:1005)
2023-09-24 10:08:00     at io.aklivity.zilla.runtime.engine/io.aklivity.zilla.runtime.engine.internal.concurent.ManyToOneRingBuffer.read(ManyToOneRingBuffer.java:181)
2023-09-24 10:08:00     at io.aklivity.zilla.runtime.engine/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.doWork(DispatchAgent.java:675)
2023-09-24 10:08:00     ... 3 more
2023-09-24 10:08:00     Suppressed: java.lang.Exception: [engine/data#3]        [0x0303000000000181] streams=[consumeAt=0x00162568 (0x0000000000162568), produceAt=0x00162568 (0x0000000000162568)]
2023-09-24 10:08:00             at io.aklivity.zilla.runtime.engine/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.doWork(DispatchAgent.java:679)
2023-09-24 10:08:00             ... 3 more
```

--- a/runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaClientConnectionPool.java
+++ b/runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaClientConnectionPool.java
@@ -1210,7 +1210,7 @@ public final class KafkaClientConnectionPool
 
             streamsByInitialIds.forEach((k, v) ->
             {
-                if(v.connection == this)
+                if (v.connection == this)
                 {
                     v.cleanupStream(traceId);
                 }

--- a/runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaClientConnectionPool.java
+++ b/runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaClientConnectionPool.java
@@ -1012,6 +1012,8 @@ public final class KafkaClientConnectionPool
                     traceId, authorization, EMPTY_EXTENSION);
 
                 state = KafkaState.closedInitial(state);
+
+                cleanupBudgetCreditorIfNecessary();
             }
         }
 
@@ -1024,6 +1026,8 @@ public final class KafkaClientConnectionPool
                     traceId, authorization, EMPTY_EXTENSION);
 
                 state = KafkaState.closedInitial(state);
+
+                cleanupBudgetCreditorIfNecessary();
             }
         }
 
@@ -1204,7 +1208,13 @@ public final class KafkaClientConnectionPool
 
             doConnectionAbort(traceId);
 
-            streamsByInitialIds.forEach((k, v) -> v.cleanupStream(traceId));
+            streamsByInitialIds.forEach((k, v) ->
+            {
+                if(v.connection == this)
+                {
+                    v.cleanupStream(traceId);
+                }
+            });
         }
 
         private void onConnectionSignal(
@@ -1311,8 +1321,6 @@ public final class KafkaClientConnectionPool
         {
             doConnectionAbort(traceId);
             doConnectionReset(traceId);
-
-            cleanupBudgetCreditorIfNecessary();
         }
 
         private void cleanupBudgetCreditorIfNecessary()

--- a/runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaClientGroupFactory.java
+++ b/runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaClientGroupFactory.java
@@ -2002,6 +2002,7 @@ public final class KafkaClientGroupFactory extends KafkaClientSaslHandshaker imp
             }
 
             cleanupEncodeSlotIfNecessary();
+            cleanupBudgetIfNecessary();
         }
 
         private void doNetworkAbort(
@@ -2015,6 +2016,7 @@ public final class KafkaClientGroupFactory extends KafkaClientSaslHandshaker imp
             }
 
             cleanupEncodeSlotIfNecessary();
+            cleanupBudgetIfNecessary();
         }
 
         private void doNetworkReset(
@@ -2382,6 +2384,15 @@ public final class KafkaClientGroupFactory extends KafkaClientSaslHandshaker imp
                 encodeSlotTraceId = 0;
             }
         }
+
+        private void cleanupBudgetIfNecessary()
+        {
+            if (initialDebIndex != NO_DEBITOR_INDEX)
+            {
+                initialDeb.release(initialDebIndex, initialBudgetId);
+                initialDebIndex = NO_DEBITOR_INDEX;
+            }
+        }
     }
 
     private final class DescribeClient extends KafkaSaslClient
@@ -2709,6 +2720,7 @@ public final class KafkaClientGroupFactory extends KafkaClientSaslHandshaker imp
             state = KafkaState.closedInitial(state);
 
             cleanupEncodeSlotIfNecessary();
+            cleanupBudgetIfNecessary();
 
             doEnd(network, originId, routedId, initialId, initialSeq, initialAck, initialMax,
                 traceId, authorization, EMPTY_EXTENSION);
@@ -2725,6 +2737,7 @@ public final class KafkaClientGroupFactory extends KafkaClientSaslHandshaker imp
             }
 
             cleanupEncodeSlotIfNecessary();
+            cleanupBudgetIfNecessary();
         }
 
         private void doNetworkReset(
@@ -3094,6 +3107,15 @@ public final class KafkaClientGroupFactory extends KafkaClientSaslHandshaker imp
                 encodeSlotTraceId = 0;
             }
         }
+
+        private void cleanupBudgetIfNecessary()
+        {
+            if (initialDebIndex != NO_DEBITOR_INDEX)
+            {
+                initialDeb.release(initialDebIndex, initialBudgetId);
+                initialDebIndex = NO_DEBITOR_INDEX;
+            }
+        }
     }
 
     private final class CoordinatorClient extends KafkaSaslClient
@@ -3444,6 +3466,7 @@ public final class KafkaClientGroupFactory extends KafkaClientSaslHandshaker imp
             }
 
             cleanupEncodeSlotIfNecessary();
+            cleanupBudgetIfNecessary();
 
         }
 
@@ -3459,6 +3482,7 @@ public final class KafkaClientGroupFactory extends KafkaClientSaslHandshaker imp
             }
 
             cleanupEncodeSlotIfNecessary();
+            cleanupBudgetIfNecessary();
         }
 
         private void doNetworkReset(
@@ -4387,6 +4411,15 @@ public final class KafkaClientGroupFactory extends KafkaClientSaslHandshaker imp
                 encodeSlot = NO_SLOT;
                 encodeSlotOffset = 0;
                 encodeSlotTraceId = 0;
+            }
+        }
+
+        private void cleanupBudgetIfNecessary()
+        {
+            if (initialDebIndex != NO_DEBITOR_INDEX)
+            {
+                initialDeb.release(initialDebIndex, initialBudgetId);
+                initialDebIndex = NO_DEBITOR_INDEX;
             }
         }
     }

--- a/runtime/binding-kafka/src/test/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/ClientGroupIT.java
+++ b/runtime/binding-kafka/src/test/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/ClientGroupIT.java
@@ -60,6 +60,16 @@ public class ClientGroupIT
     @Test
     @Configuration("client.yaml")
     @Specification({
+        "${app}/client.sent.write.abort.after.sync.group.response/client",
+        "${net}/client.sent.write.abort.after.sync.group.response/server"})
+    public void shouldHandleClientSentWriteAbortAfterSyncGroupResponse() throws Exception
+    {
+        k3po.finish();
+    }
+
+    @Test
+    @Configuration("client.yaml")
+    @Specification({
         "${app}/rebalance.protocol.highlander/client",
         "${net}/rebalance.protocol.highlander/server"})
     public void shouldLeaveGroupOnGroupRebalanceError() throws Exception

--- a/specs/binding-kafka.spec/src/main/scripts/io/aklivity/zilla/specs/binding/kafka/streams/application/group/client.sent.write.abort.after.sync.group.response/client.rpt
+++ b/specs/binding-kafka.spec/src/main/scripts/io/aklivity/zilla/specs/binding/kafka/streams/application/group/client.sent.write.abort.after.sync.group.response/client.rpt
@@ -1,0 +1,56 @@
+#
+# Copyright 2021-2023 Aklivity Inc.
+#
+# Aklivity licenses this file to you under the Apache License,
+# version 2.0 (the "License"); you may not use this file except in compliance
+# with the License. You may obtain a copy of the License at:
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+
+connect "zilla://streams/app0"
+    option zilla:window 8192
+    option zilla:transmission "half-duplex"
+
+write zilla:begin.ext ${kafka:beginEx()
+                               .typeId(zilla:id("kafka"))
+                               .group()
+                                   .groupId("test")
+                                   .protocol("highlander")
+                                   .timeout(45000)
+                                   .build()
+                               .build()}
+
+connected
+
+read zilla:begin.ext ${kafka:matchBeginEx()
+                               .typeId(zilla:id("kafka"))
+                               .group()
+                                   .groupId("test")
+                                   .protocol("highlander")
+                                   .timeout(30000)
+                                   .build()
+                               .build()}
+
+read advised zilla:flush ${kafka:flushEx()
+                             .typeId(zilla:id("kafka"))
+                             .group()
+                                 .leaderId("memberId-1")
+                                 .memberId("memberId-1")
+                                 .members("memberId-1")
+                                 .build()
+                             .build()}
+
+write zilla:data.empty
+write flush
+
+read zilla:data.empty
+
+write aborted
+read aborted

--- a/specs/binding-kafka.spec/src/main/scripts/io/aklivity/zilla/specs/binding/kafka/streams/application/group/client.sent.write.abort.after.sync.group.response/server.rpt
+++ b/specs/binding-kafka.spec/src/main/scripts/io/aklivity/zilla/specs/binding/kafka/streams/application/group/client.sent.write.abort.after.sync.group.response/server.rpt
@@ -1,0 +1,61 @@
+#
+# Copyright 2021-2023 Aklivity Inc.
+#
+# Aklivity licenses this file to you under the Apache License,
+# version 2.0 (the "License"); you may not use this file except in compliance
+# with the License. You may obtain a copy of the License at:
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+
+property serverAddress "zilla://streams/app0"
+
+accept ${serverAddress}
+    option zilla:window 8192
+    option zilla:transmission "half-duplex"
+
+accepted
+
+read zilla:begin.ext ${kafka:matchBeginEx()
+                               .typeId(zilla:id("kafka"))
+                               .group()
+                                   .groupId("test")
+                                   .protocol("highlander")
+                                   .timeout(45000)
+                                   .build()
+                               .build()}
+
+connected
+
+write zilla:begin.ext ${kafka:beginEx()
+                               .typeId(zilla:id("kafka"))
+                               .group()
+                                   .groupId("test")
+                                   .protocol("highlander")
+                                   .timeout(30000)
+                                   .build()
+                               .build()}
+write flush
+
+write advise zilla:flush ${kafka:flushEx()
+                             .typeId(zilla:id("kafka"))
+                             .group()
+                                 .leaderId("memberId-1")
+                                 .memberId("memberId-1")
+                                 .members("memberId-1")
+                                 .build()
+                             .build()}
+
+read zilla:data.empty
+
+write zilla:data.empty
+write flush
+
+read abort
+write abort

--- a/specs/binding-kafka.spec/src/main/scripts/io/aklivity/zilla/specs/binding/kafka/streams/network/group.f1.j5.s3.l3.h3/client.sent.write.abort.after.sync.group.response/client.rpt
+++ b/specs/binding-kafka.spec/src/main/scripts/io/aklivity/zilla/specs/binding/kafka/streams/network/group.f1.j5.s3.l3.h3/client.sent.write.abort.after.sync.group.response/client.rpt
@@ -1,0 +1,173 @@
+#
+# Copyright 2021-2023 Aklivity Inc.
+#
+# Aklivity licenses this file to you under the Apache License,
+# version 2.0 (the "License"); you may not use this file except in compliance
+# with the License. You may obtain a copy of the License at:
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+
+property networkConnectWindow 8192
+property instanceId ${kafka:randomBytes(42)}
+
+property newRequestId ${kafka:newRequestId()}
+property fetchWaitMax 500
+property fetchBytesMax 65535
+property partitionBytesMax 8192
+
+connect "zilla://streams/net0"
+  option zilla:window ${networkConnectWindow}
+  option zilla:transmission "duplex"
+  option zilla:byteorder "network"
+
+connected
+
+write  22                                # size
+       10s                               # find coordinator
+       1s                                # v1
+       ${newRequestId}
+       5s "zilla"                        # client id
+       4s "test"                         # "session" coordinator key
+       [0x00]                            # coordinator group type
+
+read  35                                 # size
+      (int:newRequestId)
+      0                                  # throttle time
+      0s                                 # no error
+      4s "none"                          # error message none
+      0                                  # coordinator node
+      9s "localhost"                     # host
+      9092                               # port
+
+write 82                                                 # size
+      32s                                                # describe configs
+      0s                                                 # v0
+      ${newRequestId}
+      -1s                                                # no client id
+      1                                                  # resources
+        [0x04]                                           # broker resource
+        1s "0"                                           # "node" topic
+        2                                                # configs
+          28s "group.min.session.timeout.ms"             # name
+          28s "group.max.session.timeout.ms"             # name
+
+read 103                                                # size
+      (int:newRequestId)
+      0
+      1                                                  # resources
+        0s                                               # no error
+        -1s                                              # error message
+        [0x04]                                           # broker resource
+        1s "0"                                           # "0" nodeId
+        2                                                # configs
+          28s "group.min.session.timeout.ms"             # name
+          4s "6000"                                      # value
+          [0x00]                                          # not read only
+          [0x00]                                          # not default
+          [0x00]                                          # not sensitive
+          28s "group.max.session.timeout.ms"             # name
+          5s "30000"                                     # value
+          [0x00]                                          # not read only
+          [0x00]                                          # not default
+          [0x00]                                          # not sensitive
+
+read aborted
+write abort
+
+read notify ROUTED_CLUSTER_SERVER
+
+connect await ROUTED_CLUSTER_SERVER
+        "zilla://streams/net0"
+  option zilla:window ${networkConnectWindow}
+  option zilla:transmission "duplex"
+  option zilla:byteorder "network"
+
+connected
+
+write 119                               # size
+      11s                               # join group
+      5s                                # v5
+      ${newRequestId}
+      5s "zilla"                        # client id
+      4s "test"                         # consumer group
+      30000                             # session timeout
+      4000                              # rebalance timeout
+      0s                                # consumer group member
+      42s ${instanceId}                 # group instance id
+      8s "consumer"                     # protocol type
+      1                                 # group protocol
+        10s "highlander"                     # protocol name
+        14                                   # metadata size
+        ${kafka:randomBytes(14)}             # metadata
+
+read  34                                                # size
+      (int:newRequestId)
+      0                                                 # throttle time
+      79s                                               # member id required
+      -1                                                # generated id
+      0s                                                # protocol name
+      0s                                                # leader id
+      10s "memberId-1"                                  # consumer member group id
+      0                                                 # members
+
+write 129                               # size
+      11s                               # join group
+      5s                                # v5
+      ${newRequestId}
+      5s "zilla"                        # no client id
+      4s "test"                         # consumer group
+      30000                             # session timeout
+      4000                              # rebalance timeout
+      10s "memberId-1"                  # consumer group member
+      42s ${instanceId}                 # group instance id
+      8s "consumer"                     # protocol type
+      1                                 # group protocol
+        10s "highlander"                     # protocol name
+        14                                   # metadata size
+        ${kafka:randomBytes(14)}             # metadata
+
+read  128                                               # size
+      (int:newRequestId)
+      0                                                 # throttle time
+      0s                                                # no error
+      3                                                 # generated id
+      10s "highlander"                                  # protocol name
+      10s "memberId-1"                                  # leader id
+      10s "memberId-1"                                  # consumer member group id
+      1                                                 # members
+         10s "memberId-1"                                  # consumer member group id
+         42s [0..42]                                       # group instance id
+         14                                                # metadata size
+         2s                                                # version
+         0                                                 # topics
+         0                                                 # userdata
+         0                                                 # partitions
+
+write 101                                            # size
+      14s                                           # sync group
+      3s                                            # v3
+      ${newRequestId}
+      5s "zilla"                                    # no client id
+      4s "test"                                     # consumer group
+      3                                             # generation id
+      10s "memberId-1"                              # consumer member group id
+      42s ${instanceId}                             # group instance id
+      1                                             # assignments
+        10s "memberId-1"                             # consumer member group id
+        0                                            # metadata
+
+read  14                                                # size
+      (int:newRequestId)
+      0                                                 # throttle time
+      0s                                                # no error
+      0                                                 # assignment
+
+read aborted
+write abort

--- a/specs/binding-kafka.spec/src/main/scripts/io/aklivity/zilla/specs/binding/kafka/streams/network/group.f1.j5.s3.l3.h3/client.sent.write.abort.after.sync.group.response/server.rpt
+++ b/specs/binding-kafka.spec/src/main/scripts/io/aklivity/zilla/specs/binding/kafka/streams/network/group.f1.j5.s3.l3.h3/client.sent.write.abort.after.sync.group.response/server.rpt
@@ -1,0 +1,164 @@
+#
+# Copyright 2021-2023 Aklivity Inc.
+#
+# Aklivity licenses this file to you under the Apache License,
+# version 2.0 (the "License"); you may not use this file except in compliance
+# with the License. You may obtain a copy of the License at:
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+
+property instanceId ${kafka:randomBytes(42)}
+
+accept "zilla://streams/net0"
+  option zilla:window 8192
+  option zilla:transmission "duplex"
+  option zilla:byteorder "network"
+
+accepted
+
+connected
+
+read  22                                # size
+      10s                               # find coordinator
+      1s                                # v1
+      (int:newRequestId)
+      5s "zilla"                        # client id
+      4s "test"                         # "test" coordinator key
+      [0x00]                            # coordinator group type
+
+write 35                                 # size
+      ${newRequestId}
+      0                                  # throttle time
+      0s                                 # no error
+      4s "none"                          # error message none
+      0                                  # coordinator node
+      9s "localhost"                  # host
+      9092                            # port
+
+read 82                                                 # size
+     32s                                                # describe configs
+     0s                                                 # v0
+     (int:requestId)
+     -1s                                                # no client id
+     1                                                  # resources
+       [0x04]                                           # broker resource
+       1s "0"                                           # "node" topic
+       2                                                # configs
+         28s "group.min.session.timeout.ms"             # name
+         28s "group.max.session.timeout.ms"             # name
+
+write 103                                                # size
+      ${requestId}
+      0
+      1                                                  # resources
+        0s                                               # no error
+        -1s                                              # error message
+        [0x04]                                           # broker resource
+        1s "0"                                           # "0" nodeId
+        2                                                # configs
+          28s "group.min.session.timeout.ms"             # name
+          4s "6000"                                      # value
+          [0x00]                                          # not read only
+          [0x00]                                          # not default
+          [0x00]                                          # not sensitive
+          28s "group.max.session.timeout.ms"             # name
+          5s "30000"                                     # value
+          [0x00]                                          # not read only
+          [0x00]                                          # not default
+          [0x00]                                          # not sensitive
+
+write abort
+read aborted
+
+accepted
+
+connected
+
+read  119                               # size
+      11s                               # join group
+      5s                                # v5
+      (int:newRequestId)
+      5s "zilla"                        # client id
+      4s "test"                         # consumer group
+      30000                             # session timeout
+      4000                              # rebalance timeout
+      0s                                # consumer group member
+      42s [0..42]                       # group instance id
+      8s "consumer"                     # protocol type
+      1                                 # group protocol
+        10s "highlander"                     # protocol name
+        14                                   # metadata size
+        [0..14]                              # metadata
+
+
+write 34                                                # size
+      ${newRequestId}
+      0                                                 # throttle time
+      79s                                               # member id required
+      -1                                                # generated id
+      0s                                                # protocol name
+      0s                                                # leader id
+      10s "memberId-1"                                  # consumer member group id
+      0                                                 # members
+
+read  129                               # size
+      11s                               # join group
+      5s                                # v5
+      (int:newRequestId)
+      5s "zilla"                        # no client id
+      4s "test"                         # consumer group
+      30000                             # session timeout
+      4000                              # rebalance timeout
+      10s "memberId-1"                  # consumer group member
+      42s [0..42]                       # group instance id
+      8s "consumer"                     # protocol type
+      1                                 # group protocol
+        10s "highlander"                    # protocol name
+        14                                   # metadata size
+        [0..14]                              # metadata
+
+write 128                                               # size
+      ${newRequestId}
+      0                                                 # throttle time
+      0s                                                # no error
+      3                                                 # generated id
+      10s "highlander"                                  # protocol name
+      10s "memberId-1"                                  # leader id
+      10s "memberId-1"                                  # consumer member group id
+      1                                                 # members
+         10s "memberId-1"                                  # consumer member group id
+         42s ${instanceId}                                 # group instance id
+         14                                                # metadata size
+         2s                                                # version
+         0                                                 # topics
+         0                                                 # userdata
+         0                                                 # partitions
+
+read 101                                           # size
+     14s                                           # sync group
+     3s                                            # v3
+     (int:newRequestId)
+     5s "zilla"                                    # no client id
+     4s "test"                                     # consumer group
+     3                                             # generation id
+     10s "memberId-1"                              # consumer member group id
+     42s [0..42]                                   # group instance id
+     1                                             # assignments
+       10s "memberId-1"                             # consumer member group id
+       0                                            # metadata
+
+write 14                                                # size
+      ${newRequestId}
+      0                                                 # throttle time
+      0s                                                # no error
+      0                                                 # assignment
+
+write abort
+read aborted

--- a/specs/binding-kafka.spec/src/test/java/io/aklivity/zilla/specs/binding/kafka/streams/application/GroupIT.java
+++ b/specs/binding-kafka.spec/src/test/java/io/aklivity/zilla/specs/binding/kafka/streams/application/GroupIT.java
@@ -56,6 +56,15 @@ public class GroupIT
 
     @Test
     @Specification({
+        "${app}/client.sent.write.abort.after.sync.group.response/client",
+        "${app}/client.sent.write.abort.after.sync.group.response/server"})
+    public void shouldHandleClientSentWriteAbortAfterSyncGroupResponse() throws Exception
+    {
+        k3po.finish();
+    }
+
+    @Test
+    @Specification({
         "${app}/leader/client",
         "${app}/leader/server"})
     public void shouldBecameLeader() throws Exception

--- a/specs/binding-kafka.spec/src/test/java/io/aklivity/zilla/specs/binding/kafka/streams/network/GroupIT.java
+++ b/specs/binding-kafka.spec/src/test/java/io/aklivity/zilla/specs/binding/kafka/streams/network/GroupIT.java
@@ -56,6 +56,15 @@ public class GroupIT
 
     @Test
     @Specification({
+        "${net}/client.sent.write.abort.after.sync.group.response/client",
+        "${net}/client.sent.write.abort.after.sync.group.response/server"})
+    public void shouldHandleClientSentWriteAbortAfterSyncGroupResponse() throws Exception
+    {
+        k3po.finish();
+    }
+
+    @Test
+    @Specification({
         "${net}/coordinator.not.available/client",
         "${net}/coordinator.not.available/server"})
     public void shouldHandleCoordinatorNotAvailableError() throws Exception

